lib/vector-buffers/src/topology/builder.rs

```diff
@@ -133,39 +132,17 @@ impl<T: Bufferable> TopologyBuilder<T> {
             };
 
             // Create the buffer usage handle for this stage and initialize it as we create the
-            // sender/receiver/acker.  This is slightly awkward since we just end up actually giving
+            // sender/receiver.  This is slightly awkward since we just end up actually giving
             // the handle to the `BufferSender`/`BufferReceiver` wrappers, but that's the price we
             // have to pay for letting each stage function in an opaque way when wrapped.
             let usage_handle = buffer_usage.add_stage(stage_idx);
             let provides_instrumentation = stage.untransformed.provides_instrumentation();
-            let (sender, receiver, acker) = stage
+            let (sender, receiver) = stage
                 .untransformed
                 .into_buffer_parts(usage_handle.clone())
                 .await
                 .context(FailedToBuildStageSnafu { stage_idx })?;
 
-            // Multiple components with "segmented" acknowledgements cannot be supported at the
-            // moment.  Segmented acknowledgements refers to stages which split the
-            // acknowledgement of a single event into two parts.
-            //
-            // As an example, the an in-memory stage would simply pass through an acknowledgement, as the event
-            // itself flows through untouched.  Other stages, like the disk stage, have to
-            // acknowledge an event when it is written to disk, as the acknowledgement data cannot
-            // be serialized to disk and rehydrated on deserialization.  However, the buffer still
-            // supports acknowledgments on the read side so that sinks can tell the buffer when a
-            // particular event in the buffer is safe to delete from disk, etc.
-            //
-            // In this way, the acknowledgements of an event for a disk buffer are "segmented".
-            // Since we don't have the information to track which stage in a topology has emitted an
-            // event to apply acknowledgements in the correct order, we don't support those
-            // configurations.
-            //
-            // In the future, we may opt to support such a configuration.
-            if current_acker.is_some() && acker.is_some() {
-                return Err(TopologyError::StackedAcks);
-            }
-            current_acker = acker;
```

Comment:
I'm trying to think through if we're breaking an invariant here without some sort of compensating code... and also a little mad at myself I didn't explain in the comment why segmented acknowledgement can't be supported i.e. what it would actually break.

Reply:
Did you come to any conclusions about issues here?

Reply:
Thinking it through, I understand why I originally said this: we only supported ordered acknowledgements via `Driver` or by doing so manually in a given sink. This meant that if two stages in a buffer topology acknowledged records between _each other_, we'd have to ensure that they also did so in order, which practically speaking, we already achieve since buffers require mutable access to write, which provides the serialized ordering.

However, that was never a given in terms of internal buffer behavior, and acknowledgements could have _potentially_ happened out of order, hence the comment/error.

Now that we've eschewed `Acker`, and force the ordering of acknowledgement via finalization, any buffer implementation which would normally handle acknowledgements in order can do so whether it's the only stage in the buffer topology, or one of many stages.